CRM/Core/Payment/Manual.php

```diff
@@ -111,8 +113,8 @@ public function getPaymentFormFields() {
    * @throws \Civi\Payment\Exception\PaymentProcessorException
    */
   public function doPayment(&$params, $component = 'contribute') {
-    $params['payment_status_id'] = $this->getResult();
-    return $params;
+    $params = PropertyBag::cast($params);
+    return ['payment_status_id' => $this->getResult()];
```

Comment:
@eileenmcnaughton The results array will *only* contain a single value with this change - `payment_status_id`. That's not what we want to happen.

Reply:
I know they won't get set with the manual processor but things like `trxn_id` and `fee_amount` are often set within `doPayment()` (retrieved as part of the transaction with the processor) and will not be available before the function is called.

Reply:
@mattwire  - right so there are 2 parts of this 

1) the manual processor - this only sets one value so we only need to return one
2) other processors - with this change anything that is returned from the function is still used - however, on the off-chance processors are adding a value to $params & not to the return array it could get lost - I'm gonna add some handling for that for the 3 relevant fields

Note 1 & 2 could be separate PRs - 2 would have to happen first

Reply:
@mattwire I've added handling+ a deprecation notice for fee_amount. On checking the code it was clear that if processors were not returning trxn_id & payment_status_id they could expect unreliable results so we can assume they are not doing that.

I did not add any handling for total_amount. It's really unclear how the form layer would be expected to cope with total_amount being altered by the processor so I don't think we should see the possibility of someone doing that as supported (although this would be really unlikely to break that hack anyway since processors generally return whatever they have for $result anyway.